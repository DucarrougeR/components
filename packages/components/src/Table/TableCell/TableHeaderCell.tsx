import { CompatibleHTMLProps } from '@looker/design-tokens'
import styled from 'styled-components'
import { TableCellProps, tableCellCSS, tableCellDefaults } from './tableCell'

export interface TableHeaderCellProps
  extends TableCellProps,
    CompatibleHTMLProps<HTMLTableHeaderCellElement> {}

export const TableHeaderCell = styled.th<TableHeaderCellProps>`
  ${tableCellCSS}
`

TableHeaderCell.defaultProps = {
  ...tableCellDefaults,
  color: 'palette.charcoal400',
  fontSize: 'xsmall',
  fontWeight: 'semiBold',
}
