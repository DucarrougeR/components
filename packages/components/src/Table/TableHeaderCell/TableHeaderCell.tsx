import React, { FunctionComponent } from 'react'
import styled from 'styled-components'
import { Box, BoxProps } from '../../Box'

export type TableHeaderCellProps = BoxProps<HTMLTableHeaderCellElement>
type ComponentType = FunctionComponent<TableHeaderCellProps>

const InternalTableHeaderCell: ComponentType = props => (
  <Box
    // @ts-ignore
    as={'th' as 'th'}
    py="xsmall"
    fontSize="xsmall"
    color="palette.charcoal400"
    fontWeight="semiBold"
    {...props}
  />
)

export const TableHeaderCell = styled<ComponentType>(InternalTableHeaderCell)``
