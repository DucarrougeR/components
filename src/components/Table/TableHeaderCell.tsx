import styled from '../../styled_components'
import { gray400 } from '../../styles/colors'

export const TableHeaderCell = styled('th')`
  padding: ${props => props.theme.spacing.s} 0;
  font-size: font-size(6);
  color: ${gray400};
  font-weight: 600;
`
