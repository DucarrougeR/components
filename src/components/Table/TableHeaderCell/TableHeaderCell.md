```js noeditor
const StatusAndResources = require('../../../../styleguide_components/StatusAndResources').StatusAndResources;

<StatusAndResources
  status="experimental"
  figmaURL="https://www.figma.com/file/TIVgrnhEWwRGvr5yEQA4CVlj/Tables"
  githubURL="https://github.com/looker/relens/blob/master/src/components/Table/TableHeaderCell/TableHeaderCell.tsx"
  feedbackTitle="TableHeaderCell Component Feedback" />
```
