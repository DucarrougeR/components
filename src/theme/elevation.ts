export type ElevationLayers = '1' | '2' | '3' | '4' | '5' | '6' | '7' | 'max'

export type Shadows = { [K in ElevationLayers]?: string }

export const shadows: Shadows = {
  1: '0 1px 2px rgba(0, 0, 0, 0.1), 0px 1px 1px rgba(0, 0, 0, 0.05)',
  2: '0 3px 4px rgba(0, 0, 0, 0.11), 0 1px 2px rgba(0, 0, 0, 0.04)',
  3: '0 6px 10px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.04)',
  4: '0 9px 18px rgba(0, 0, 0, 0.14), 0 1px 2px rgba(0, 0, 0, 0.04),  0 10px 8px 4px rgba(0, 0, 0, 0.01)',
  5: '0 14px 20px rgba(0, 0, 0,.15), 0 1px 1px rgba(0, 0, 0, 0.04),  0 14px 10px 8px rgba(0, 0, 0, 0.02)',
  6: '0 22px 42px 0px rgba(0, 0, 0,.18),  0 1px 1px rgba(0, 0, 0, 0.05), 0 16px 20px 10px rgba(0, 0, 0, 0.025)',
}
