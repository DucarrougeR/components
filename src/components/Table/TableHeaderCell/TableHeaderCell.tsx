import * as React from 'react'
import { styled } from '../../../style'
import { Box, BoxProps } from '../../Box'

const InternalTableHeaderCell: React.FC<
  BoxProps<HTMLTableHeaderCellElement>
> = ({ ...props }) => (
  <Box
    is="th"
    py="xsmall"
    fontSize="xsmall"
    color="palette.charcoal400"
    fontWeight="semiBold"
    {...props}
  />
)

export const TableHeaderCell = styled(InternalTableHeaderCell)``
