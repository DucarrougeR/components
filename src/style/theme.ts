import { rem } from 'polished'
import { InterpolationValue } from 'styled-components'
import {
  SizeLarge,
  SizeMedium,
  SizeNone,
  SizeSmall,
  SizeXLarge,
  SizeXSmall,
  SizeXXLarge,
  SizeXXXLarge,
} from '../types'
import { Easings, easings } from './easings'
import { shadows, Shadows } from './elevation'
import { fontFaces, FontFaces } from './font_faces'
import { FontRamp, fontSizes, lineHeights } from './font_sizes'
import { fontWeights } from './font_weights'
import { palette, Palette } from './palette'
import { SemanticColors, semanticColors } from './semantic_colors'
import { transitions, Transitions } from './transitions'

export * from './easings'
export * from './elevation'
export * from './font_sizes'
export * from './palette'
export * from './semantic_colors'
export * from './transitions'

export type SpacingSizes =
  | SizeNone
  | SizeLarge
  | SizeMedium
  | SizeSmall
  | SizeXLarge
  | SizeXSmall
  | SizeXXLarge
  | SizeXXXLarge

export interface Theme {
  breakpoints: string[]
  easings: Easings
  fontFaces: FontFaces
  fontSizes: FontRamp
  fontWeights: Record<string, number>
  lineHeights: FontRamp
  palette: Palette
  /**
   * A function that can be overridden to return different reset css properties
   * or null to remove all resets. Most base elements in Lens implement the reset.
   */
  reset: () => InterpolationValue
  semanticColors: SemanticColors
  shadows: Shadows
  space: Record<SpacingSizes, string>
  transitions: Transitions
}

export const theme: Theme = {
  breakpoints: ['30rem', '48rem', '64rem', '75rem', '90rem'],
  easings,
  fontFaces,
  fontSizes,
  fontWeights,
  lineHeights,
  palette,
  reset: () => {
    return {
      border: 0,
      font: 'inherit',
      fontSize: '100%',
      margin: 0,
      padding: 0,
      verticalAlign: 'baseline',
    }
  },
  semanticColors,
  shadows,
  space: {
    // tslint:disable:object-literal-sort-keys
    none: 'none',
    xsmall: rem(4),
    small: rem(8),
    medium: rem(16),
    large: rem(20),
    xlarge: rem(32),
    xxlarge: rem(40),
    xxxlarge: rem(60),
    // tslint:enable:object-literal-sort-keys
  },
  transitions,
}
