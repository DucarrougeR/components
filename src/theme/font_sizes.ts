import { rem } from 'polished'
import { RampSizes } from '../style/ramp_sizes'

export type FontRamp = { [K in RampSizes]: string }

// tslint:disable:object-literal-sort-keys
export const fontRamp: FontRamp = {
  d1: rem(58),
  d2: rem(46),
  d3: rem(36),
  1: rem(25),
  2: rem(22),
  3: rem(18),
  4: rem(16),
  5: rem(14),
  6: rem(12),
}

// tslint:disable:object-literal-sort-keys
export const lineHeightRamp: FontRamp = {
  d1: rem(84),
  d2: rem(64),
  d3: rem(52),
  1: rem(40),
  2: rem(32),
  3: rem(28),
  4: rem(24),
  5: rem(20),
  6: rem(16),
}
