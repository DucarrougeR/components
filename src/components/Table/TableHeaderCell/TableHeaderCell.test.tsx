import 'jest-styled-components'
import * as React from 'react'
import { create } from 'react-test-renderer'

import theme from '../../../theme'
import { TableHeaderCell } from './TableHeaderCell'

test('A <TableHeaderCell> should render', () => {
  const component = create(<TableHeaderCell theme={theme}>1</TableHeaderCell>)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
