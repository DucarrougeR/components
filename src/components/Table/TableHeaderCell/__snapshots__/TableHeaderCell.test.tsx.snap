// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A <TableHeaderCell> should render 1`] = `
.c1 {
  margin: 0;
  padding: 0;
  border: 0;
  font: inherit;
  font-size: 100%;
  vertical-align: baseline;
}

.c0 {
  padding: 0.5rem 0;
  font-size: font-size(6);
  color: #939BA5;
  font-weight: 600;
}

<th
  className="c0 c1"
>
  1
</th>
`;
